scripts/run_bon_v2.py

```diff
@@ -0,0 +1,357 @@
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+# Runs best of n (BoN) ranking
+# TODO: implement this for DPO models
+
+import argparse
+import logging
+import os
+import sys
+
+import numpy as np
+import torch
+import transformers
+from accelerate import Accelerator
+from accelerate.logging import get_logger
+from fastchat.conversation import get_conv_template
+from tqdm import tqdm
+from transformers import AutoTokenizer, pipeline
+
+from rewardbench import (
+    REWARD_MODEL_CONFIG,
+    check_tokenizer_chat_template,
+    load_bon_dataset_v2,
+    reroll_and_score_dataset,
+    save_to_hub,
+)
+
+# from rewardbench.constants import EXAMPLE_COUNTS, SUBSET_MAPPING
+# from rewardbench.utils import calculate_scores_per_section
+
+# get token from HF_TOKEN env variable, but if it doesn't exist pass none
+HF_TOKEN = os.getenv("HF_TOKEN", None)
+# this is necessary to automatically log in when running this script in docker/batch beaker jobs
+if HF_TOKEN is not None:
+    from huggingface_hub._login import _login
+
+    _login(token=HF_TOKEN, add_to_git_credential=False)
+
+
+def get_args():
+    """
+    Parse arguments strings model and chat_template
+    """
+    parser = argparse.ArgumentParser()
+    parser.add_argument("--model", type=str, required=True, help="path to model")
+    parser.add_argument(
+        "--dataset", type=str, required=True, help="dataset, both .jsonl (local) and huggingface format supported"
+    )
+    parser.add_argument("--tokenizer", type=str, default=None, help="path to non-matching tokenizer to model")
+    parser.add_argument("--chat_template", type=str, default="tulu", help="path to chat template")
+    parser.add_argument(
+        "--trust_remote_code", action="store_true", default=False, help="directly load model instead of pipeline"
+    )
+    parser.add_argument("--do_not_save", action="store_true", help="do not save results to hub (for debugging)")
+    parser.add_argument("--batch_size", type=int, default=64, help="batch size for inference")
+    parser.add_argument("--max_length", type=int, default=2048, help="Max length of RM inputs (passed to pipeline)")
+    parser.add_argument("--best_of", type=int, default=4, help="number of best of n to select from")
+    parser.add_argument(
+        "--pref_sets", action="store_true", help="run on common preference sets instead of our custom eval set"
+    )
+    parser.add_argument(
+        "--debug", action="store_true", help="run on common preference sets instead of our custom eval set"
+    )
+    parser.add_argument(
+        "--disable_beaker_save", action="store_true", help="disable saving the main results in a file for AI2 Beaker"
+    )
+    parser.add_argument(
+        "--not_quantized", action="store_true", help="disable quantization for models that are quantized by default"
+    )
```

Comment:
For v2 lets make not quantized default? Have been lots of questions on it and I don't know the implementation robustness